src/test/libcephfs/test.cc

```diff
@@ -1425,16 +1425,24 @@ TEST(LibCephFS, SlashDotDot) {
   /* At root, "." and ".." should be the same inode */
   ASSERT_EQ(ino, st.st_ino);
 
-
-  /* Make sure it works same way when mounting subtree */
+  /* Test accessing the parent of an unlinked directory */
   char dir1[32], dir2[32];
   sprintf(dir1, "/sldotdot%x", getpid());
   sprintf(dir2, "%s/sub%x", dir1, getpid());
 
   ASSERT_EQ(ceph_mkdir(cmount, dir1, 0755), 0);
   ASSERT_EQ(ceph_mkdir(cmount, dir2, 0755), 0);
+
+  ASSERT_EQ(ceph_chdir(cmount, dir2), 0);
+  ASSERT_EQ(ceph_rmdir(cmount, dir2), 0);
+  ASSERT_EQ(ceph_stat(cmount, dir1, &st), 0);
+  ino = st.st_ino;
+  ASSERT_EQ(ceph_stat(cmount, "..", &st), -ENOENT);
```

Comment:
Because dir2 is now unlinked. We chdir into "dir2", then rmdir "dir2" and then see whether the parent of that dir is now accessible. It's not, so you get -ENOENT back (no parent).

Hmm...come to think of it -- maybe we do need to handle the readdir case here as well? No reason you couldn't issue a readdir from in that now-unlinked directory.